ldes-server-infra-mongo/mongo-retention-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/MemberPropertiesRepositoryImpl.java

```diff
@@ -101,4 +112,85 @@ public void removeMemberPropertiesOfCollection(String collectionName) {
 	public void deleteById(String id) {
 		memberPropertiesEntityRepository.deleteById(id);
 	}
+
+	@Override
+	public Stream<MemberProperties> findExpiredMemberProperties(ViewName viewName,
+																TimeBasedRetentionPolicy policy) {
+		return memberPropertiesEntityRepository
+				.findMemberPropertiesEntitiesByCollectionNameAndViewsContainingAndTimestampBefore(
+						viewName.getCollectionName(),
+						viewName.asString(),
+						LocalDateTime.now().minus(policy.duration())
+				)
+				.map(memberPropertiesEntityMapper::toMemberProperties);
+	}
+
+	@Override
+	public Stream<MemberProperties> findExpiredMemberProperties(ViewName viewName,
+																VersionBasedRetentionPolicy policy) {
+		int versionsToKeep = policy.numberOfMembersToKeep();
+		String collectionName = viewName.getCollectionName();
+		String viewNameAsString = viewName.asString();
+
+		final SortOperation sort = sortTimestampDesc();
+		final MatchOperation match =
+				match(Criteria.where(COLLECTION).is(collectionName).and(VIEWS).in(viewNameAsString));
+		final GroupOperation group = groupOnVersionOf();
+		final ProjectionOperation project = projectVersionsToKeep(versionsToKeep);
+		final UnwindOperation unwind = unwindDocuments();
+		final ReplaceRootOperation replaceRoot = getReplaceRootDocuments();
+
+		final Aggregation aggregation = createAggregation(
+				sort, match, group, project, unwind, replaceRoot
+		);
+
+		return mongoTemplate.aggregateStream(aggregation, MemberPropertiesEntity.NAME, MemberProperties.class);
+	}
+
+	@Override
+	public Stream<MemberProperties> findExpiredMemberProperties(ViewName viewName,
+																TimeAndVersionBasedRetentionPolicy policy) {
+		final MatchOperation match =
+				match(Criteria
+						.where(COLLECTION).is(viewName.getCollectionName())
+						.and(VIEWS).in(viewName.asString())
+						.and(TIMESTAMP).lt(LocalDateTime.now().minus(policy.duration())));
+
+		final Aggregation aggregation = createAggregation(
+				sortTimestampDesc(),
+				match,
+				groupOnVersionOf(),
+				projectVersionsToKeep(policy.numberOfMembersToKeep()),
+				unwindDocuments(),
+				getReplaceRootDocuments()
+		);
+
+		return mongoTemplate.aggregateStream(aggregation, MemberPropertiesEntity.NAME, MemberProperties.class);
+	}
+
+	private Aggregation createAggregation(AggregationOperation... operations) {
+		final AggregationOptions aggregationOptions = AggregationOptions.builder().allowDiskUse(true).build();
+		return newAggregation(operations).withOptions(aggregationOptions);
+	}
+
+	private ReplaceRootOperation getReplaceRootDocuments() {
+		return replaceRoot(DOCUMENTS);
+	}
+
+	private UnwindOperation unwindDocuments() {
+		return unwind(DOCUMENTS);
+	}
```

Comment:
would you be able to document what these methods do? To me, these methods are clearly mongo operations, but would require a person to first know it's mongodb specific and then hope to find the correct documentation online.

I know this is a bit of nitpicking.. but this seems very non business code

Reply:
These are just private helper methods where we call a mongo unwind (https://www.mongodb.com/docs/manual/reference/operator/aggregation/unwind/).. Since this is the mongo implementation, we should be able to expect people to understand or at least be able to consult mongo documentation.